x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/CompositeRolesStore.java

```diff
@@ -421,4 +409,39 @@ void merge(MergeableIndicesPrivilege other) {
             }
         }
     }
+
+    private static final class RolesRetrievalResult {
+
+        private final Set<RoleDescriptor> roleDescriptors = new HashSet<>();
+        private Set<String> missingRoles = Collections.emptySet();
+        private boolean failure = false;
+
+        private void addDescriptors(Set<RoleDescriptor> descriptors) {
+            roleDescriptors.addAll(descriptors);
+        }
+
+        private Set<RoleDescriptor> getRoleDescriptors() {
+            return roleDescriptors;
+        }
+
+        private void setFailure() {
+            failure = true;
+        }
+
+        private boolean isFailure() {
```

Comment:
It feels more natural to me to have a `success` with a default true value and an `isSuccessful()` method. No concrete arguments for this, just personal sense of how consumable/readable that is, so take if with a grain of salt.